src/components/TextField/TextField.tsx

```diff
@@ -20,12 +21,13 @@ export const TextField: React.FC<Props> = ({
   required = false,
   onChange = () => {},
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
   const [touched, setToched] = useState(false);
   const hasError = touched && required && !value;
+  const hasValidateError = touched && value && !pattern.test(value) && (
+    label === 'Image URL' || label === 'Imdb URL'
+  );
```

Comment:
```suggestion
  const hasValidateError = (label === 'Image URL' || label === 'Imdb URL')
    && touched
    && value
    && !pattern.test(value);
```

Order it properly to prevent unnecessary calculations (on falsy value in will return immediately)

Reply:
I'm not sure it's good to use label names as condition :harold: Use data attributes or something more stable.